pkg/controller/security/manager.go

```diff
@@ -283,5 +220,10 @@ func (s *SecretManager) rotateCert(identity string) {
 	}
 	s.certsCache.mu.Unlock()
 
+	if time.Until(certificate.cert.ExpireTime) >= 1*time.Hour {
+		// This can happen when delete a certificate following adding the same one later.
+		log.Debugf("cert %s expire at %T, skip rotate now", identity, certificate.cert.ExpireTime)
+	}
```

Comment:
How is `workqueue.DelayingInterface` used? Can it accept duplicate elements? I don't fully understand the case how this scenario occured.

Reply:
WorkQueue doesnot store duplicate elements, but as we are using delay push, there can be duplicate elmenets delay pushes. And as long as one is consumed, the other can be consumed later.

Reply:
WorkQueue is indeed very convenient. 
I have no other questions.

Reply:
What is the progress of your unit test, could this one be merged first?